public/locations-list-map.js

```diff
@@ -328,10 +395,188 @@ function initMap(stateFilter) {
+ */
+function centerMapToMarkersNearUser() {
+  // First check to see if the user will accept getting their location, if not, silently return
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function (position) {
+      // Use navigator provided lat/long coords to center map now.
+      centerMapToMarkersNearCoords(position.coords.latitude, position.coords.longitude);
+
+    }, function (err) {
+      // Hide the "User my location" link since we know that will not work.
+      $('#use-location').hide();
+
+    }, {
+      maximumAge: Infinity,
+      timeout: 10000
+    });
+  }
+}
+
+
+
+/**
+ * Centers map around markers nearest to an arbitrary set of latitude/longitude coordinates.
+ */
+function centerMapToMarkersNearCoords(latitude, longitude) {
+  var latlng = new google.maps.LatLng(latitude, longitude);
+
+  //Compute the distances of all markers from the user
+  var markerDistances = new Map(); // an associative array containing the marker referenced by the computed distance
+  var distances = []; // all the distances, so we can sort and then call markerDistances
+  for (const marker of markers) {
+    let distance = google.maps.geometry.spherical.computeDistanceBetween(marker.position, latlng);
+
+    // HACK: In the unlikely event that the exact same distance is computed, add one meter to the distance to give it a unique distance
+    // This could occur if a marker was added twice to the same location.
+    if (markerDistances.has(distance)) { distance = distance + 1; }
+
+    markerDistances[distance] = marker;
+
+    distances.push(distance);
+  }
+
+  // sort the distances and set bounds to closest three
+  distances.sort((a, b) => a - b);
+
+  // center the map on the user
+  const bounds = new google.maps.LatLngBounds();
+  bounds.extend(latlng);
+
+  // Extend the bounds to contain the three closest markers
+  for(let i = 0; i < 3; i++) {
+    // Get one of the closest markers
+    let distance = distances[i];
+    let marker = markerDistances[distance];
+
+    // Add to the iterator first just in case something fails later to avoid infinite loop
+    i++;
+
+    const marker_lat = marker.position.lat();
+    const marker_lng = marker.position.lng();
+
+    const loc = new google.maps.LatLng(marker_lat, marker_lng);
+    bounds.extend(loc);
+  }
+  map.fitBounds(bounds);       // auto-zoom
+  map.panToBounds(bounds);     // auto-center
+}
+
+/********************************
+ * END MAP SEARCH FUNCTIONALITY *
+ ********************************/
+
+
+
+/**
+ * Changes the markers currently rendered on the map based strictly on . This will reset the 'markers' module variable as well.
+ *
+ * TODO (patricknelson): The param "data" is redundant and should be removed! Currently it only refers to the global "data_by_location".
+ *
+ * TODO (patricknelson): The param "showNearest" is used only once and is not relevant to the purpose of this method (showing specific markers only depending on filters).
```

Comment:
Please remove this comment. It's clutter for now. If you want to track a bunch of these cleanups, put them in a single issue and then submit focused PRs.